src/app/overview/overview.component.ts

```diff
@@ -35,14 +36,21 @@ export class OverviewComponent implements OnInit {
     this.initiator$.subscribe(() => {
       this.programName = this.storage.getUser().programName;
       this.fastFeedbackService.pullFastFeedback().subscribe();
+      this.dismissPermissionModal();
     });
   }
 
   async checkPNPermission(snapshot: RouterStateSnapshot): Promise<void> {
-    const showPrompt = await this.pushNotificationService.promptForPermission(PermissionTypes.firstVisit, snapshot);
+    const showPrompt =  await this.pushNotificationService.promptForPermission(PermissionTypes.firstVisit, snapshot);
     if (showPrompt) {
       await this.notificationService.pushNotificationPermissionPopUp('Would you like to receive  push notification?', 'assets/img/permissions.svg');
     }
     return;
   }
-}
+
+  dismissPermissionModal (){
+    const stateChange =  this.pushNotificationService.listenToStateChangeToActive();
+    if (stateChange ) {
```

Comment:
@ronakamani seems like you missed the checking of whether the push notification permission has granted.

From the Jira story, the criteria are these:

- if we can listen to the event, if we found user has updated and permitted push notification permission to "allowed" from "disapproved", we dismiss the popup for them, or
- if we detect user still don't permit Push Notification, we let user to dismiss it manually.

Reply:
oh sorry, it was by accident. Thanks @trtshen .